luxonis_train/attached_modules/metrics/ocr_accuracy.py

```diff
@@ -0,0 +1,109 @@
+import logging
+
+import torch
+from torch import Tensor
+
+from luxonis_train.utils import Labels, Packet, prepare_batch_targets
+
+from .base_metric import BaseMetric
+
+logger = logging.getLogger(__name__)
+
+
+class OCRAccuracy(
+    BaseMetric[list[dict[str, Tensor]], list[dict[str, Tensor]]]
+):
+    """Accuracy metric for OCR tasks."""
+
+    def __init__(self, blank_cls: int = 0, **kwargs):
+        """Initializes the OCR accuracy metric.
+
+        @type blank_cls: int
+        @param blank_cls: Index of the blank class. Defaults to C{0}.
+        """
+        super().__init__(**kwargs)
+        self.blank_cls = blank_cls
+        self._init_metric()
+
+    def _init_metric(self):
+        """Initializes the running metric."""
+        self.running_metric = {
+            "acc_0": 0.0,
+            "acc_1": 0.0,
+            "acc_2": 0.0,
+        }
+        self.n = 0
+
+    def prepare(
+        self, inputs: Packet[Tensor], labels: Labels
+    ) -> tuple[Tensor, Tensor]:
+        """Prepares the predictions and targets for accuracy
+        computation.
+
+        @type inputs: Packet[Tensor]
+        @param inputs: A packet containing input tensors, typically
+            network predictions.
+        @type labels: Labels
+        @param labels: A dictionary containing text labels and
+            corresponding lengths.
+        @rtype: tuple[Tensor, Tensor]
+        @return: A tuple of predictions and targets.
+        """
+
+        preds = inputs["/classification"][0]
+        targets_batch = labels["/metadata/text"]
+        target_lengths = labels["/metadata/text_length"].int()
+
+        targets = prepare_batch_targets(targets_batch, target_lengths)
+        targets = self.node.encoder(targets).to(preds.device)  # type: ignore
```

Comment:
Instead of `# type: ignore`, you can add a class attribute annotation `node: OCRTHead`. This will make the types work and also it will check on the background that the module is connected to a compatible node.

This applies to the visualizer and loss as well.